### ISSUE TITLE
Case where a secondary user needs to edit a record

### ISSUE THREAD

**Issue by AmberSJones** [UCHIC/CKANDev#92]
We need to investigate whether there is a straightforward option to give a user that was not the original submitter and not an admin access to edit a submission. Perhaps this functionality is possible via Groups? I'm not familiar with what functionality is available with Groups.

**Comment by pkdash** [comment]
It seems CKAN "Group" feature is for grouping a number of datasets on a common name for easy access. The "group" has nothing to do with the access permission.

**Comment by AmberSJones** [comment]
Ok- thanks. Is there a different way we can provide access to another user(s)?

**Comment by pkdash** [comment]
Do we want a person with 'editor' role to be able to edit a private dataset? Is that the requirement? Then I can ask the CKAN forum for some possible solution.

**Comment by AmberSJones** [comment]
Yes, an editor needs to be able to edit a dataset that they didn't upload- but only certain datasets- maybe the creator or an admin gives access somehow?

**Comment by pkdash** [comment]
It looks like we have to write some code to have this functionality. First of all we have to think about how the admin will assign edit permission to certain users with editor role for editing a specific private dataset, It looks like it has to be done in the datatset editing page where the admin can click a button (Add Editor) and then select form a list of users with editor role. This information will be saved as extra metadata for that dataset when the admin updates the dataset. Then I have to implement an authorization function for the 'package_update' CKAN action to see if the current user is authorized to edit the datatset or not. I am assuming this should work but not 100% sure though. This will take at least a week to implement and test.

**Comment by AmberSJones** [comment]
I think the workflow that @pkdash outlines sounds reasonable. Let's discuss tomorrow if we should proceed with it.

**Comment by AmberSJones** [comment]
We've decided not to pursue any implementation at this point, however, if in the future we want to address this need, it may best be done via Groups where any member of a group can edit the datasets belonging to that group.

**Comment by AmberSJones** [comment]
This functionality has been requested again. We need to re-visit discussion of these options- perhaps at our next meeting.

**Comment by pkdash** [comment]
Sounds good.

**Comment by pkdash** [comment]
@AmberSJones This seems to be a bigger task which may take at least a week of work to add this new feature. So this needs to be done at a later time.
